Pidgin.Tests/ParserTestBase.cs

```diff
@@ -19,5 +19,13 @@ public class ParserTestBase
             Assert.Equal(errorPos, result.Error.ErrorPos);
             Assert.Equal(consumedInput, result.ConsumedInput);
         }
+
+        protected void AssertUnexpectedFailure<TToken, T>(Result<TToken, T> result, Maybe<TToken> unexpected, SourcePos errorPos, bool consumedInput)
+        {
+            Assert.False(result.Success);
+            Assert.Equal(unexpected, result.Error.Unexpected);
+            Assert.Equal(errorPos, result.Error.ErrorPos);
+            Assert.Equal(consumedInput, result.ConsumedInput);
+        }
```

Comment:
The vast majority of failures have both an `Unexpected` token and some `Expected` input. It means "here's what I was hoping for, but here's what I actually got". Perhaps the terminology could do with a little finessing, they do sound like "types of error" rather than properties of all errors.